src/conv/heuristics/ai_heuristics.cpp

```diff
@@ -616,9 +618,19 @@ bool ModelSetParams(const std::string& arch,
         dim = std::sqrt(features.size());
     else
         dim = features.size();
+    auto start             = std::chrono::high_resolution_clock::now();
     fdeep::tensors context = model->Encode(features, dim, transform_features);
     float decoder_input    = 0.0;
-    for(std::size_t i = 0; i < model->metadata.num_tuning_params; ++i)
+    std::string dir;
+    switch(direction)
+    {
+    case miopen::conv::Direction::Forward: dir = "fwd"; break;
+    case miopen::conv::Direction::BackwardData: dir = "bwd"; break;
+    case miopen::conv::Direction::BackwardWeights: dir = "wrw"; break;
+    default: return false;
+    }
+    std::size_t num_tuning_params = model->metadata.num_tuning_params[dir];
```

Comment:
In `src/kernels/gfx908_ConvAsm1x1U_metadata.ktn.model`, only "fwd" is present under "num_tuning_parameters". So if the GPU is gfx908 and the solver is ConvAsm1x1U and `direction` is `miopen::conv::Direction::BackwardData`, line 632 would error out

Reply:
Both "fwd" and "bwd" are defined in the metadata file.

Reply:
What about `wrw`? Also, src/kernels/gfx90a_ConvHipIgemmGroupFwdXdlops_metadata.ktn.model only has `fwd`

Reply:
ConvHipIgemmGroupFwdXdlops is only a forward direction solver. ConvAsm1x1U solver is not applicable for backward weights (https://github.com/ROCm/MIOpen/pull/2898/files#diff-d69d92a5ae6963da140aa45d0ec4ef7317f368a93fbc6076c783142ef3be7d82L535).